### PR TITLE
Metrics testing

### PR DESCRIPTION
**Pull request by pzel**
This branch introduces a `instrumented_story`, via which the test author can specify how MongooseIM metrics should change after a given story is run.
As part of the change, some new tests were added and the metric coverage substantially increased.

### CODE REVIEW

File: tests/metrics_api_SUITE.erl [hunk 1]
```diff
@@ -67,108 +73,158 @@ message_flow(Config) ->
...
-        end).
-
-sessions(Config) ->
-    escalus:story(Config, [{alice, 2}, {bob, 1}], fun(_Alice1, _Alice2, _Bob) ->
-        3 = fetch_global_counter_value(totalSessionCount, Config),
-        2 = fetch_global_counter_value(uniqueSessionCount, Config)
-    end).
+
+one_client_just_logs_in(Config) ->
+    instrumented_story
```
Comment (michalwski) [inline tests/metrics_api_SUITE.erl]: Team, do you find this intuitive? For me the opening parenthesis should be in the same line as the function name. What we have here is quite odd to me.

Comment (lavrin) [inline tests/metrics_api_SUITE.erl]: I would do it like:

```erlang
S = fun (JarekKrolWiertarek) ->
        ...
    end.
instrumented_story(..., S).
```
Breaking line just after the opening parentheses looks ugly (my personal opinion).

**Comment by lavrin** [comment]
Tests passed - esl/MongooseIM#380 - merging in.

File: tests/metrics_api_SUITE.erl (hunk 1 above)
Comment (michalwski) [inline tests/metrics_api_SUITE.erl]: :+1: this would be most readable.

**Pull request merged**